src/internal/IntersectBSplines.cpp

```diff
@@ -311,8 +311,8 @@ namespace
 
             gp_Vec diff = p1.XYZ() - p2.XYZ();
             F = diff.SquareMagnitude();
-            G(1) = 2. * diff.Dot(d1)* (m_c1->LastParameter()-m_c1->FirstParameter()) * d_getUParam(X.Value(1));
-            G(2) = -2. * diff.Dot(d2) * (m_c2->LastParameter()-m_c2->FirstParameter()) * d_getUParam(X.Value(2));
+            G(1) = 2. * diff.Dot(d1) * d_getUParam(X.Value(1));
+            G(2) = -2. * diff.Dot(d2) * d_getUParam(X.Value(2));
```

Comment:
```suggestion
            G(2) = -2. * diff.Dot(d2) * d_getVParam(X.Value(2));
```

Hi @gongfan99 . I think it must be also d_getVParam, right? Here we are computing the derivative to v (the second parameter). Can you please verify. I think you are right regarding the removal  of the (last_parm- first_parm) factor.

Reply:
Yes, you are correct.